### PR TITLE
Reference for Intl.DurationFormat

### PR DESCRIPTION
**Pull request by romulocintra**
#### Summary

MDN Docs for https://github.com/tc39/proposal-intl-duration-format

#### Supporting details

- https://bugzilla.mozilla.org/show_bug.cgi?id=1648139
- https://bugs.chromium.org/p/v8/issues/detail?id=11660
- https://bugs.webkit.org/show_bug.cgi?id=214794

#### Related issues

https://github.com/tc39/ecma402-mdn/issues/22

#### Metadata

- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by wbamberg** [comment]
@romulocintra , sorry to be so slow getting to this PR. It looks like this isn't implemented anywhere yet, so I'm not sure we are ready to have docs on MDN for it (also it's hard, for example, for me to check that the examples work properly).

Perhaps it would be best to reopen this PR when it is implemented in at least one browser?

**Comment by romulocintra** [comment]
> @romulocintra , sorry to be so slow getting to this PR. It looks like this isn't implemented anywhere yet, so I'm not sure we are ready to have docs on MDN for it (also it's hard, for example, for me to check that the examples work properly).
> 
> Perhaps it would be best to reopen this PR when it is implemented in at least one browser?

Converted it to Draft, as soon we have an implementation will re-open it. thanks @wbamberg 

cc @sffc

File: files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      - : The style to be used for formatting milliseconds.
+        - "`long`"
+        - "`short`"
+        - "`narrow`"
+        - "`numeric`"
+
+    - `millisecondsDisplay`
+      - : Whether to always display milliseconds, or only if nonzero.
+        - "`always`"
+        - "`auto`"
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md:149]:
```suggestion
        - `"always"`
        - `"auto"`
```

File: files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+    - `microseconds`
+      - : The style to be used for formatting microseconds.
+        - "`long`"
+        - "`short`"
+        - "`narrow`"
+        - "`numeric`"
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md:156]:
```suggestion
        - `"long"`
        - `"short"`
        - `"narrow"`
        - `"numeric"`
```

File: files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+    - `microsecondsDisplay`
+      - : Whether to always display microseconds, or only if nonzero.
+        - "`always`"
+        - "`auto`"
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md:161]:
```suggestion
        - `"always"`
        - `"auto"`
```

File: files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+    - `nanoseconds`
+      - : The style to be used for formatting nanoseconds.
+        - "`long`"
+        - "`short`"
+        - "`narrow`"
+        - "`numeric`"
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md:168]:
```suggestion
        - `"long"`
        - `"short"`
        - `"narrow"`
        - `"numeric"`
```

File: files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+    - `nanosecondsDisplay`
+      - : Whether to always display nanoseconds, or only if nonzero.
+        - "`always`"
+        - "`auto`"
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/intl/durationformat/durationformat/index.md:173]:
```suggestion
        - `"always"`
        - `"auto"`
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Nice job!

I'm so excited to see this work getting closer to being ready to be merged as soon as a browser implements this feature. So, I went ahead with a first pass of reviews.

To save you time, I crafted suggestions for most comments:

- These are primarily typographic corrections.
- I also fixed some details to prevent our linters (Markdownlint, Prettier) from complaining.
- I removed `Tags:` as we removed them from the whole of MDN since the first version of this PR.
- I added `page-type:` instead, as this is what replaced them.

There will be a second pass later, but that way, all these typographic thingies won't get in the way.

Thanks a lot for your work.

**Review by Josh-Cena — APPROVED** [review]
This is implemented in Safari 16.4, and these docs look comparable to other `Intl` pages in terms of structure and content, so I think it's near ready to go.

**Review by teoli2003 — APPROVED** [review]

**Comment by teoli2003** [comment]
@romulocintra When you bring this PR out of draft (meaning you think it is ready), we can merge it. Note that in case, we can always continue work in follow-ups.

**Comment by teoli2003** [comment]
\o/

**Comment by romulocintra** [comment]
> \o/

Thank you for following along, this was quite long journey thank you again

**Comment by Elchi3** [comment]
Is there no `supportedLocalesOf` here like there is for the other Intl APIs?

**Comment by romulocintra** [comment]
> Is there no `supportedLocalesOf` here like there is for the other Intl APIs?

Yes, good catch, there is I'll prepare PR upcoming days for it - https://tc39.es/proposal-intl-duration-format/#sec-Intl.DurationFormat.supportedLocalesOf

**Pull request merged**